### ISSUE TITLE
clang-tidy modernize-concat-nested-namespaces false positive when using an attribute with a string that contains a double colon.

### ISSUE THREAD

**Issue by kees-jan** [llvm/llvm-project#57530]
Thanks for clang-tidy. It's saving me a truckload of work! Also, I apologize for the long, and probably not very clear title.

In my `/tmp` directory, I have a file `nested-namespaces.cc` containing:
```C++
namespace [[deprecated("::")]] name {
}
```
There's also a `compile_commands.json` reading:
```json
[{
  "directory": "/tmp",
  "command": "clang++ -c --std=c++17 /tmp/nested-namespaces.cc",
  "file": "/tmp/nested-namespaces.cc"
}]
```
I'm running the command
```shell
clang-tidy -p=/tmp --checks='-*,modernize-concat-nested-namespaces' nested-namespaces.cc
```

Expected result: No warnings
Actual result:
```console
1 warning generated.
/tmp/nested-namespaces.cc:3:1: warning: nested namespaces can be concatenated [modernize-concat-nested-namespaces]
namespace [[deprecated("::")]] name {
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
namespace name
```
When asking clang-tidy to fix the issue, the deprecated attribute is removed. The result compiles, but is obviously not what I'd expect

This is with clang-tidy 16.0.0 taken from `apt.llvm.org/focal` (on ubuntu 20.04, obviously), though older versions were affected as well

**Comment by kees-jan** [comment]
The [files](https://github.com/llvm/llvm-project/files/9479973/files.zip) I used, for your convenience
